src/Orleans.Core.Abstractions/IDs/ClientGrainId.cs

```diff
@@ -45,12 +47,11 @@ public static bool TryParse(GrainId grainId, out ClientGrainId clientId)
             }
 
             // Strip the observer id, if present.
-            var key = grainId.Key.ToStringUtf8();
-            if (key.IndexOf(ObserverGrainId.SegmentSeparator) is int index && index >= 0)
+            var key = grainId.Key.AsSpan();
+            if (key.IndexOf((byte)ObserverGrainId.SegmentSeparator) is int index && index >= 0)
```

Comment:
I initially thought that the SegmentSeparator byte may be allowed to exist in a valid UTF8 sequence which doesn't include the SegmentSeparator char, but I think that's untrue. Still, it would be good to be certain.

Reply:
It's safe to search for any ASCII character in an UTF-8 sequence by just the byte value.

Reply:
Sounds good. I took a closer look, and I agree. That's a nice artifact of the encoding.